staging/src/k8s.io/apiserver/pkg/server/storage/resource_config.go

```diff
@@ -85,22 +117,25 @@ func (o *ResourceConfig) removeMatchingResourcePreferences(matcher func(gvr sche
 func (o *ResourceConfig) DisableVersions(versions ...schema.GroupVersion) {
 	for _, version := range versions {
 		o.GroupVersionConfigs[version] = false
-
 		// a preference about a version takes priority over the previously set resources
 		o.removeMatchingResourcePreferences(resourceMatcherForVersion(version))
 	}
 }
 
 // EnableVersions enables all resources in a given groupVersion.
+// A groupVersion can only be enabled if its APILifecyle is available for the emulation version.
 // This will remove any preferences previously set on individual resources.
 func (o *ResourceConfig) EnableVersions(versions ...schema.GroupVersion) {
 	for _, version := range versions {
-		o.GroupVersionConfigs[version] = true
-
+		if available, err := o.versionAvailable(version); available {
+			o.GroupVersionConfigs[version] = true
+		} else {
+			o.GroupVersionConfigs[version] = false
+			klog.V(1).Infof("version %s cannot be enabled because: %s", version.String(), err.Error())
+		}
```

Comment:
Shouldn't this be feature gated? (Ideally we minimize execution into new code paths when the feature gate is off, this seems like a place we could do that?)

Reply:
it is feature gated in`versionAvailable`